src/com/perl5/lang/perl/idea/refactoring/extract/PerlExtractMethodHandler.java

```diff
@@ -0,0 +1,139 @@
+/*
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.perl5.lang.perl.idea.refactoring.extract;
+
+import com.intellij.codeInsight.codeFragment.CodeFragment;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.SelectionModel;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.refactoring.RefactoringActionHandler;
+import com.intellij.refactoring.extractMethod.AbstractExtractMethodDialog;
+import com.intellij.refactoring.extractMethod.ExtractMethodDecorator;
+import com.intellij.refactoring.extractMethod.ExtractMethodValidator;
+import com.intellij.refactoring.util.AbstractVariableData;
+import com.intellij.util.containers.ArrayListSet;
+import com.perl5.lang.perl.fileTypes.PerlFileTypeScript;
+import com.perl5.lang.perl.idea.refactoring.PerlRefactorUtils;
+import com.perl5.lang.perl.idea.refactoring.PerlRefactor;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import java.util.Set;
+
+public class PerlExtractMethodHandler implements RefactoringActionHandler {
+
+  @Override
+  public void invoke(@NotNull Project project, Editor editor, PsiFile file, DataContext dataContext) {
+
+    final SelectionModel selectionModel = editor.getSelectionModel();
```

Comment:
No need to final everything, it's Java 8

Reply:
Sorry not familiar with Java 8 changes yet. Is final not needed at all anymore or just in some cases?

Reply:
Can't remember i've used it. Java 8 infers final very good.